src/Listener/UnapproveNewContent.php

```diff
@@ -44,34 +45,45 @@ public function approveByDefault(ConfigureModelDefaultAttributes $event)
     public function unapproveNewPosts(PostWillBeSaved $event)
     {
         $post = $event->post;
+        $unapprove = false;
 
         if (! $post->exists) {
             $ability = $post->discussion->number_index == 0 ? 'startWithoutApproval' : 'replyWithoutApproval';
 
-            if ($event->actor->can($ability, $post->discussion)) {
-                if ($post->is_approved === null) {
-                    $post->is_approved = true;
-                }
+            $disallowedTags = Tag::getIdsWhereCannot($event->actor, 'discussion.' . $ability);
 
-                return;
+            foreach ($event->data['relationships']['tags']['data'] as $value) {
+                if (in_array($value['id'], $disallowedTags)) {
+                    $unapprove = true;
+                }
             }
 
-            $post->is_approved = false;
+            if($unapprove === false) {
```

Comment:
That said, `$unapprove` also implies a negative. Can we call this `$approve` and switch the values around maybe?